internal/api/maps_place_search.go

```diff
@@ -0,0 +1,173 @@
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"time"
+
+	"github.com/gin-gonic/gin"
+
+	"github.com/photoprism/photoprism/internal/auth/acl"
+	"github.com/photoprism/photoprism/internal/event"
+	"github.com/photoprism/photoprism/pkg/txt"
+)
+
+type PlaceSearchResult struct {
```

Comment:
Ideally, these structs should be imported from the service/hub/places package, or a new package should be created for them. Also, public types require documentation, e.g. "PlaceSearchResult represents place search result data."